src/apis/Style.res

```diff
@@ -81,20 +154,146 @@ type position = [#absolute | #relative]
 // because layout styles props don't accept animated values.
 // We don't do the distinction as ReScript is an HMTS that doesn't support implicit subtyping
 
-// ____DangerouslyImpreciseStyle_Internal
-@obj
-external // Dangerous Imprecise Style
+// Dangerous Imprecise Style
 // Contains all of
 // - image style
 // - text style
 // - view style
 // - transform style
 // - shadow style
 // - layout style
+type styleObj = {
```

Comment:
Just a question here for everyday need:

Now we are supposed to do `style={Style.styleObj({width: 10.->Style.dp})}` instead of  `style={Style.style(~width=10.->Style.dp,())}`. Is there really a direct benefit ? Because with style being styleObj, at the end, we don't save a single char :D 
Could we imagine something shorter ? Or maybe directly a record ? I know that array would be an issue but... What if we only accept array as discussed in #632 ? This way, we could only type ` style={[{width: 10.->dp}]}`. 
I have to admit that with themed app (light/dark mode) and predefined common styles, I have most of the time use array.

And it would be like React.useState() for example. Original api accepts value or function, but rescript bindings only accept the "most advanced" to avoid bloated api. By using array, we directly allow us to save chars everytime !

Any thoughts ? cc @Freddy03h

Reply:
Yes, no `~` anymore, no `()` at the end anymore that you can forget, more readable and more familiar to JS developers, definitely a benefit in any case.

Reply:
But it would certainly be nice to be able to avoid `Style.style`/`Style.styleObj` altogether.

Reply:
I don't see how it could work in a type-safe manner though, given that nested style arrays are also allowed (and sometimes useful), and that we may also want to keep `arrayOption`. I think we do need our abstract type.

Reply:
Maybe we should just make it `s` instead of `styleObj`. Example:

```rescript
let styles = {
  open Style
  StyleSheet.create({
    "playButton": s({
      display: #flex,
      alignItems: #center,
      justifyContent: #center,
      borderWidth: StyleSheet.hairlineWidth,
    }),
  })
}
```

Reply:
For me, the best seems to be with array : `style={[{width: 10.->dp}]}`
The point of using a record is to not using a function and to be easier for JS dev.

I never needed or seen nested arrays for style in all apps I worked on, so I don't understand which cases are useful.

My concern is about `arrayOption`, should we'll do thing like this?
```res
style={[{width: 10.->dp}->Some, optionStyleFromProp]->keepMap(s => s)}
```

Reply:
An alternative to `arrayOption` can be to use an empty style (`Style.style()`) instead of `None`.
We could even provide a pre-created one (`Style.empty`) for this purpose.

Reply:
Yes, or maybe there could be a dedicated helper function `orEmpty`:

```rescript
style={[{width: 10.->dp}, optionStyleFromProp->orEmpty]}
```

Reply:
I'm fine with this, even with `Option.getWithDefault(Style.empty)`, it's less annoying than adding `Some` to every other styles when transforming a `Style.array` to `Style.arrayOption`, like currently.

Reply:
I think nested array or not a real problem. Users can flatten array themselves (eg: `Belt.Array.concatMany`). So I won't bother with this.

And for optional style, solution like `orEmpty` is the good choice. Will make things easier.
We could even name it `optional`, `optionalStyle` or `optionStyle` to be explicit.

Worst case scenario could be:

```rescript
style={Belt.Array.concatMany([[{width: 10.->dp}, userStyle->optional], arrayOfStyles])}
```

If that's properly documented, it's imo, the way to go !

Reply:
look great!

Reply:
You two are slowly convincing me. 🙂
I will update the PR accordingly.

Reply:
♥️

Reply:
Done!

With all these breaking changes I guess we'll need a codemod?